### ISSUE TITLE
Error when downloading mps-youtube through pip3

### ISSUE THREAD

**Issue by wandbrandon** [mps-youtube/mps-youtube#432]
» sudo pip3 install mps-youtube  
Collecting mps-youtube
Requirement already satisfied (use --upgrade to upgrade): pafy!=0.4.0,!=0.4.1,!=0.4.2,>=0.3.82 in /usr/local/lib/python3.5/dist-packages (from mps-youtube)
Requirement already satisfied (use --upgrade to upgrade): youtube-dl in /usr/local/lib/python3.5/dist-packages (from pafy!=0.4.0,!=0.4.1,!=0.4.2,>=0.3.82->mps-youtube)
Installing collected packages: mps-youtube
Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/basecommand.py", line 209, in main
    status = self.run(options, args)
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/req/req_set.py", line 731, in install
    **kwargs
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/req/req_install.py", line 841, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/req/req_install.py", line 1040, in move_wheel_files
    isolated=self.isolated,
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/wheel.py", line 375, in move_wheel_files
    clobber(source, dest, False, fixer=fixer, filter=filter)
  File "/usr/local/lib/python3.5/dist-packages/pip-8.0.2-py3.5.egg/pip/wheel.py", line 321, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/usr/lib/python3.5/shutil.py", line 115, in copyfile
    with open(dst, 'wb') as fdst:
NotADirectoryError: [Errno 20] Not a directory: '/usr/local/share/applications/mps-youtube.desktop'

can anyone help me solve this problem? it says pafy and youtube-dl are up to date

**Comment by hrnr** [comment]
ups desktop file installation now fails. maybe changes in python3.5 wheels?

**Comment by wandbrandon** [comment]
I'm sorry I'm kind of new to all of this, you mind explaining that a bit?

**Comment by hrnr** [comment]
tldr;
try:
```
sudo mkdir -p /usr/local/share/applications
```

Sure, my apologies. From the traceback problem occurs when pip is installing desktop file to '/usr/local/share/applications/mps-youtube.desktop'. This location seems fine, so path was resolved properly.

Problem occured when pip tried to copy desktop file (install it). NotADirectoryError might be a bit misleading in this case. I think it is just complaining some directory in the path is not existing. You might want to create it (probably with the same permissions as /usr/share/applications)?

I'm not sure if creating that directory should be reponsibility of python installer (if so, than probably bug in python installer?). Surely it should hadle failure gracefully.

BTW: the desktop file is only required for playmenu integration on ubuntu, but we provide desktop file for all linuxes as it is freedesktop standard for desktop app and shouldn't couse any trouble.

Cheers,
Jiri

**Comment by wandbrandon** [comment]
Thank you so much, but it says that the directory already exists.
